src/main/java/io/cryostat/credentials/Credentials.java

```diff
@@ -41,38 +42,58 @@
-        return V2Response.json(
-                Response.Status.OK,
-                credentials.stream()
-                        .map(
-                                c -> {
-                                    try {
-                                        return Credentials.safeResult(c, targetMatcher);
-                                    } catch (ScriptException e) {
-                                        logger.warn(e);
-                                        return null;
-                                    }
-                                })
-                        .filter(Objects::nonNull)
-                        .toList());
+    public Response list() {
+        try {
+            List<Credential> credentials = Credential.listAll();
+            List<Map<String, Object>> results =
+                    credentials.stream()
+                            .map(
+                                    c -> {
+                                        try {
+                                            return Credentials.safeResult(c, targetMatcher);
+                                        } catch (ScriptException e) {
+                                            logger.warn(e);
+                                            return null;
+                                        }
+                                    })
+                            .filter(Objects::nonNull)
+                            .toList();
+
+            JsonObject response = new JsonObject();
+            response.put(
+                    "meta", new JsonObject().put("type", "application/json").put("status", "OK"));
+            response.put("data", new JsonObject().put("result", results));
+
+            return Response.ok(response.encode()).type(MediaType.APPLICATION_JSON).build();
+        } catch (Exception e) {
+            logger.error("Error listing credentials", e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+        }
     }
 
     @GET
     @RolesAllowed("read")
     @Path("/{id}")
-    public V2Response get(@RestPath long id) throws ScriptException {
-        Credential credential = Credential.find("id", id).singleResult();
-        return V2Response.json(Response.Status.OK, safeMatchedResult(credential, targetMatcher));
+    public Response get(@RestPath long id) throws ScriptException {
+        try {
+            Credential credential = Credential.find("id", id).singleResult();
+            Map<String, Object> result = safeMatchedResult(credential, targetMatcher);
+            return Response.ok(result).type(MediaType.APPLICATION_JSON).build();
+        } catch (ScriptException e) {
+            logger.error("Error retrieving credential", e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+        } catch (Exception e) {
+            logger.error("Credential not found", e);
+            return Response.status(Response.Status.NOT_FOUND).build();
```

Comment:
I think the NOT_FOUND case should already be handled here:

https://github.com/cryostatio/cryostat/blob/e6457be1994acd310cbbc03734a0c577b5ae817c/src/main/java/io/cryostat/ExceptionMappers.java#L45

Reply:
And if for some reason not, then this final catch-clause should catch something more specific than `Exception`, since there could be a lot of other kinds of unexpected RuntimeException that may occur and which should not map to a NOT_FOUND status.